src/ngraph/builder/reshape.cpp

```diff
@@ -229,7 +229,7 @@ shared_ptr<Node> builder::opset1::expand_dims(const Output<Node>& value, size_t
     return builder::opset1::reshape(value, output_shape);
 }
 
-shared_ptr<Node> builder::opset1::squeeze(const Output<Node>& value, vector<int64_t> axes)
```

Comment:
Since this is just a helper function which is intended to call with already normalized axes. It internally doesn't handle negative axes so in this way I enforce it's correct behavior. This is not an implementation of v1::Squeeze operator which you can find here: https://github.com/NervanaSystems/ngraph/blob/aosewski/group_conv_bprop_v1/src/ngraph/op/fused/squeeze.cpp#L35-L92.

Reply:
I'm not agree. Why should I care about axes normalization? As I understand builder returns V1::Squeeze op and I don't want to normalize axes. From my point of view axes normalization is only required in V1::Squeeze shape inference.
Also we have a lot of bugs in nGraph where AxisSet is used and negative values leads to undefined behavior.

Reply:
If you do not want to have bugs because of negative axes, then axes normalization is the way to go. I'm not sure how many nGraph ops has support for negative axis value, but a lot of ONNX operators can accept such values. As I said earlier this is a helper function which forces developer to not pass negative axis value, which would lead to errors.

Reply:
Moreover the axes values have to be non negative, otherwise we would get error here: https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/builder/reshape.cpp#L242. 
And currently `ngraph::Squeeze` doesn't actually support negative axes: https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/op/fused/squeeze.cpp#L47

Reply:
Yes I do not wan to have bugs. So I've created a bug for Squeeze and Unsqueeze ops as they must support negative axes according to spec.
Also talking about bugs, how you prohibit user to pass negative values to size_t argument?

Reply:
Regarding the Squeeze/Unsqueeze support for negative values, that's right, they should be fixed, however they're out of the scope of this PR. The user of this _helper function_ should be a frontend/core developer.
I do not prohibit ONNX user. The ONNX model can have negative values and we simply first normalize them in ONNX Importer code.